apps/web/src/components/borrow/table/DepositModal.tsx

```diff
@@ -0,0 +1,293 @@
+import React, {
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useState,
+  useContext,
+} from 'react'
+import {
+  BaseInput,
+  BoxLayout,
+  Button,
+  CellProps,
+  CenterLayout,
+  ColumnLayout,
+  Icon,
+  Modal,
+  StackLayout,
+  Table,
+  Text,
+} from '@bloxifi/ui'
+import { ThemeContext } from 'styled-components'
+import { BorrowAndLending, Tokens } from '@bloxifi/core'
+import { CheckAllowanceFunction } from '@bloxifi/types'
+
+import { FormattedNumber } from '../FormattedNumber'
+
+import { Web3Container } from '@/containers/Web3Container'
+import { WalletBalance } from '@/containers/WalletContainer'
+
+type Props = {
+  /**
+   * Boolean value that defines if modal is open or closed
+   */
+  isOpen?: boolean
+  /**
+   * Function that defines closing of modal
+   */
+  onClose?: () => void
+  /**
+   * Selected asset reserve data
+   */
+  data: WalletBalance
+}
+
+export const DepositModal: FunctionComponent<Props> = ({
+  isOpen,
+  onClose,
+  data,
+}: Props) => {
+  const themeContext = useContext(ThemeContext)
+  const [amountError, setAmountError] = useState<boolean>(false)
+  const [amount, setAmount] = useState<number>()
+
+  const {
+    state: { currentAccount, provider, isSupportedNetwork },
+  } = Web3Container.useContainer()
+  const signer = provider.getSigner()
+
+  const [hasError, setHasError] = useState()
+  const [loading, setLoading] = useState(false)
+
+  const [shouldApproveContract, setShouldApproveContract] = useState(false)
+  const [approved, setApproved] = useState<boolean>(false)
+  const [depositCompleted, setDepositCompleted] = useState<boolean>(false)
+  const tokenContract = Tokens.getTokenContract(signer, data.symbol)
+  const lendingPoolContract =
+    BorrowAndLending.lendingPool.getLendingPoolContract(signer)
+
+  const isApproveDisabled =
+    !isSupportedNetwork || loading || approved || !data.balance
+  const isDepositDisabled =
+    !isSupportedNetwork ||
+    loading ||
+    amountError ||
+    !amount ||
+    hasError ||
+    (shouldApproveContract && !approved)
+
+  const checkAllowance: CheckAllowanceFunction = useCallback(async () => {
+    try {
+      const approvedTokens = await Tokens.getAllowance(
+        tokenContract,
+        currentAccount,
+        'deposit',
+      )
+      setShouldApproveContract(approvedTokens.toString() === '0')
+      setHasError(null)
+    } catch (error) {
+      setHasError(error)
+    }
+  }, [currentAccount, tokenContract])
+
+  useEffect(() => {
+    if (isSupportedNetwork) {
```

Comment:
Well, we cannot fetch the data if wrong network is connected. We're getting an error: 
![image](https://user-images.githubusercontent.com/42462482/180727430-fb61f1ce-4684-472a-a3c4-c2c83e85ea31.png)

Maybe we should display something similar to "Not connected" page?

Reply:
hm, interesting, what are we passing to the graphql perhaps a user, and that's how we get an error.
because some queries are userFree (can be fetched even on the wrong network), can you check what is the data that we can have when network is wrong?